### DISCUSSION TITLE
Why this library have node version set in package.json?

### DISCUSSION THREAD

**Discussion by CaioQuirinoMedeiros** [react-hook-form/react-hook-form#8971] [Q&A] [answered]
Hey guys, sorry if it's a silly question, but can someone explain to me why this library have node version defined in package.json?

```json
"engines": {
  "node": ">=12.22.0"
}
```
This caused me problems installing the dependencies on an environment with node 12.21. I managed to install the dependencies using `--ignore-engines` and my app seems to be working fine... But I'm wondering if it might face problems because of it.

**Comment by Moshyfawn** [comment, marked as answer]
This defines the supported Node version. Opting-in to using this package with lower Node versions using the `--ignore-engines` flag doesn't guarantee the package will work in your environment as expected

**Reply by CaioQuirinoMedeiros** [reply]
I understand that it defines the supported Node version, but I wanted to know why it exists. I mean, at the end of the day, the app is just a bunch of javascript running on client's browser isn't it? So why have a minimum node version defined?

**Reply by Moshyfawn** [reply]
We wouldn't have Node versions in the first place, if there wasn't a difference in what features it supports / drops ;)

I don't have the exact reason why this specific Node version decisions was made, tho. You can take a look at the commit history to try find the specific reason #7095

**Reply by CaioQuirinoMedeiros** [reply]
Ok, I'll try to find out. Thank you @Moshyfawn!
